spec/features/global_announcements_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+RSpec.describe 'Global announcements', type: :feature do
+  subject { page }
+  let(:instance) { create(:instance) }
+
+  with_tenant(:instance) do
+    describe 'no global announcements' do
+      before do
+        instance.announcements.clear
```

Comment:
Rspec did pretty much of this for you, by nesting your tests are still isolated, there is no depending on each other and the order is random, it's actually doing the same thing as what you are doing there.

In this case it still call <code>instance.annoucements.clear</code> for 3 times, and things under describe runs in a random order.